### PR TITLE
Blob lieutenant warns before possible encounter, new ways to deal with it

### PR DESCRIPTION
**Pull request by RenechCDDA**
#### Summary
Balance "Blob lieutenant warns before possible encounter, new ways to deal with it"

#### Purpose of change
Intended to supersede #70971

#### Describe the solution
Clean up some existing EOC code.

Add a new warning variable, add logic to give at least one warning before full spawn. Vastly change the minimum and maximum time between spawn checks so that the average is about the same but warnings don't let the player run around freely at night without a worry.
-The warnings are currently split into two categories, one for early on in the cataclysm (pre-7 days, before it can spawn), and one for later on in the cataclysm. This can be expanded as need be, and since it's snippets it's pretty easy to add in more messages or change/remove ones as need be.

Remove extraneous resistances from vanilla (have not been re-added to relevant mods - I don't know them enough to balance them there. Better it be too easy to kill in mods than too hard to kill in vanilla)

Prevent `A shadow?` from appearing as a hallucination. While having it appear as a hallucination is pretty cool, it can also really, really, really break the scripting. So let's just disallow it.

You can just plain run away now. You don't need to get to cover, just far enough away is fine. Kevin suggested 10 OMTs distance, I was thinking 5. I'm splitting the difference and going with 7.5.
image
image

#### Describe alternatives you've considered
* #70971

#### Testing
JSON loads, no logic testing done as of this writing

#### Additional context
Known deficiency: Applying the blinded effect (to make the `A shadow? dies!` log message go away) still gives a log message about being blinded. And despite lasting 0 turns, the message about your sight returning takes a few turns to come back. But I feel like that is a much more tolerable message than possibly making the player be mistaken that it is permanently dead.

### CODE REVIEW

**Comment by RenechCDDA** [comment]
I talked about changing the structure of the fight in #70951 but that(the teleporting-on-hit EOC) isn't in this PR.

Still, we have at least two new (reasonable) ways to deal with it, some warnings for guard rails, and some code cleanup. I think this is looking good for one PR.

File: data/json/monsters/zed_lieutenant.json [hunk 1]
```diff
@@ -431,5 +468,22 @@
         "force": true
       }
     ]
+  },
+  {
+    "type": "snippet",
+    "category": "Shadow_Warnings_Snippets_early",
+    "text": [
+      "Even out here you can still see the burning cities lighting up the horizon.  What scares you most is not the gunshots you sometimes hear in the far distance, but the idea of what people keep shooting at.  Tonight you hear automatic gunfire winding down into single shots.\n\nJust what could make them shoot that much, all the way out here?",
+      "You suddenly realize this area seems almost devoid of life.  The few bits of blackened grass you can see are barely recognizable as grass, even discounting the night.  Sure the zombies might be tearing people apart, but what happened to the grass?  What, is the grass coming to life to eat people too?\nYou snort at the idea, but it doesn't feel safe out here."
```
Comment (Inglonias) [inline data/json/monsters/zed_lieutenant.json:477]: Gotta admit, these snippets are definitely better written than the warning I came up with.

**Comment by Inglonias** [comment]
I actually do like this better than mine with one exception - I felt really clever about the heavy-duty headlamp. I might port put that in a different PR, but it felt like a pretty simple and powerful anti-Shadow measure: Just duct taping a giant flashlight to your face.

**Comment by Zireael07** [comment]
Plz make a separate PR for the heavy duty headlamp

File: data/json/monsters/zed_lieutenant.json [hunk 2]
```diff
@@ -380,16 +406,27 @@
...
       "and": [
-        { "or": [ { "npc_allies": 3 }, { "u_at_om_location": "FACTION_CAMP_ANY" }, { "not": "u_is_outside" }, "is_day" ] },
+        { "not": { "compare_string": [ "", { "global_val": "shadow_encounter_XYZ_pos" } ] } },
+        {
+          "or": [
+            { "npc_allies": 3 },
+            { "u_at_om_location": "FACTION_CAMP_ANY" },
+            { "not": "u_is_outside" },
+            "is_day",
+            { "math": [ "distance('u', shadow_encounter_XYZ_pos)", ">=", "180" ] }
```
[the remark this answers is not in the document]
Comment (RenechCDDA) [reply data/json/monsters/zed_lieutenant.json:419]: Think you've misread that, it's already set to 180.

**Comment by Maleclypse** [comment]
Thar be conflicts fer resolven

File: data/json/effects.json [hunk 3]
```diff
@@ -629,6 +629,21 @@
     ],
     "flags": [ "EFFECT_LIMB_SCORE_MOD", "BLIND" ]
   },
+  {
+    "type": "effect_type",
```
Comment (Maleclypse) [inline data/json/effects.json:633]:
```suggestion
    "type": "effect_type"
```

File: data/json/effects.json [hunk 4]
```diff
@@ -629,6 +629,21 @@
     ],
     "flags": [ "EFFECT_LIMB_SCORE_MOD", "BLIND" ]
   },
+  {
+    "type": "effect_type"
```
Comment (Maleclypse) [inline data/json/effects.json:633]:
```suggestion
    "type": "effect_type",
```

File: data/json/effects.json [hunk 5]
```diff
@@ -629,6 +629,21 @@
     ],
     "flags": [ "EFFECT_LIMB_SCORE_MOD", "BLIND" ]
   },
+  {
+    "type": "effect_type",
+    "//1": "Copy of blind effect, used to temporarily remove vision as a utility function. Only use outside of EOC_LIEUTENANT_SHADOW_WITHDRAWS_FOR_NOW with extreme caution.",
+    "id": "blind_no_msg",
+    "name": [ "Blind" ],
+    "desc": [ "Range of Sight: 0.  You cannot see anything. You shouldn't be able to see this description either!" ],
```
Comment (Maleclypse) [inline data/json/effects.json:637]:
```suggestion
    "desc": [ "Range of Sight: 0.  You cannot see anything.  You shouldn't be able to see this description either!" ],
```

**Comment by RenechCDDA** [comment]
OK hopefully it should be good to go after the push I've just made. Unfortunate timing for the PR, been hit with merge conflicts, eoc syntax change, the works! But I'm confident we should be good now unless master is failing to build

**Pull request merged**